rendermodules/materialize/schemas.py

```diff
@@ -0,0 +1,121 @@
+
+from marshmallow import post_load, validate
+
+import argschema
+from argschema.fields import (Str, OutputDir, Int, Boolean, Float,
+                              List, InputDir, InputFile, Dict, Nested)
+from rendermodules.module.schemas import RenderClientParameters
+
+
+class MaterializedBoxParameters(argschema.schemas.DefaultSchema):
+    stack = Str(required=True, description=(
+        "stack fromw which boxes will be materialized"))
+    rootDirectory = OutputDir(required=True, description=(
+        "directory in which materialization directory structure will be "
+        "created (structure is "
+        "<rootDirectory>/<project>/<stack>/<width>x<height>/<mipMapLevel>/<z>/<row>/<col>.<fmt>)"))
+    width = Int(required=True, description=(
+        "width of flat rectangular tiles to generate"))
+    height = Int(required=True, description=(
+        "height of flat rectangular tiles to generate"))
+    maxLevel = Int(required=False, default=0, description=(
+        "maximum mipMapLevel to generate."))
+    fmt = Str(required=False, validator=validate.OneOf(['PNG', 'TIF', 'JPG']),
+              description=("image format to generate mipmaps -- "
+                           "PNG if not specified"))
+    maxOverviewWidthAndHeight = Int(required=False, description=(
+        "maximum pixel size for width or height of overview image.  "
+        "If excluded or 0, no overview generated."))
+    skipInterpolation = Boolean(required=False, description=(
+        "whether to skip interpolation (e.g. DMG data)"))
+    binaryMask = Boolean(required=False, description=(
+        "whether to use binary mask (e.g. DMG data)"))
+    label = Boolean(required=False, description=(
+        "whether to generate single color tile labels rather "
+        "than actual images"))
+    createIGrid = Boolean(required=False, description=(
+        "whther to create an IGrid file"))
+    forceGeneration = Boolean(required=False, description=(
+        "whether to regenerate existing tiles"))
+    renderGroup = Int(required=False, description=(
+        "index (1-n) identifying coarse portion of layer to render"))
+    numberOfRenderGroups = Int(required=False, description=(
+        "used in conjunction with renderGroup, total number of groups "
+        "being used"))
+
+
+class ZRangeParameters(argschema.schemas.DefaultSchema):
+    # TODO how does this work with zValues?
+    minZ = Int(required=False, description=("minimum Z integer"))
+    maxZ = Int(required=False, description=("maximum Z integer"))
+
+
+class RenderWebServiceParameters(argschema.schemas.DefaultSchema):
+    baseDataUrl = Str(required=False, description=(
+        "api endpoint url e.g. http://<host>[:port]/render-ws/v1"))
+    owner = Str(required=False, description=("owner of target collection"))
+    project = Str(required=False, description=("project fo target collection"))
+
+
+class RenderParametersRenderWebServiceParameters(RenderWebServiceParameters):
+    render = Nested(
```

Comment:
If you are talking about the post_load, I could not find one with this full functionality.  in addition, the other schemas require a full render field and just allow it to supercede the RenderWebServiceParameters.  Again, it would be good to have a single superclass which could define this for client jobs, but I think that's outside the scope of this feature branch.

Reply:
fair enough.. we should maybe add an issue to unify the RenderWebServiceParameters class and reutilize it across the board.